### PR TITLE
Support MinGW with CMake

### PR DESCRIPTION
**Pull request by akof1314**
Now supports MinGW  testcpp.

### CODE REVIEW

**Comment by dumganhar** [comment]
Travis ci build failed. Please check https://travis-ci.org/cocos2d/cocos2d-x/jobs/16643418. Thanks.

**Comment by akof1314** [comment]
@dumganhar Now is build ok.

File: cocos/2d/CCActionCamera.cpp [hunk 1]
```diff
@@ -182,11 +182,11 @@ void OrbitCamera::startWithTarget(Node *target)
 
     float r, zenith, azimuth;
     this->sphericalRadius(&r, &zenith, &azimuth);
-    if( isnan(_radius) )
+    if( std::isnan(_radius) )
```
Comment (dumganhar) [inline cocos/2d/CCActionCamera.cpp]: Why to change this to `std::isnan`. VS compiler doesn't have a `std::isnan`.

**Comment by akof1314** [comment]
@dumganhar because it conflicted with math.h isnan since c++ 11.  Now is ok.

**Comment by dumganhar** [comment]
@akof1314 Thanks. :) Are there some tutorials for using `mingw` to build cocos2d-x ?

**Comment by akof1314** [comment]
@dumganhar Yes.I will write an article for using mingw to build cocos2d-x later. (With Chinese, English is poor)

**Comment by martell** [comment]
This has broken Mingw64 support.
Mingw is really old and considered deprecated.

Mingw64 already has ssize_t amongst other variables and functions are defined and match visual studio headers much better.
I'll write a quick tutorial on how to compile now in a few minutes.

**Comment by martell** [comment]
https://gist.github.com/martell/8362294

I've created an install guide for msys2 and mingw64 for use with cocos2d-x in 6 simple steps :)

It seems the ssize_t definition breaks mingw64 right now though.
Commenting it out in CCStdC.h and CCConsole.h fixes it.
I'll have to compare the definition in libwebsockets and Mingw64 to see if they are compatable

**Comment by akof1314** [comment]
@martell It is very good!

**Comment by akof1314** [comment]
@dumganhar Cocos2d-x 3.X CMake MinGW build with Qt Creator IDE (http://blog.csdn.net/akof1314/article/details/18189069  (Chinese))

**Comment by dumganhar** [comment]
@akof1314 
I sent you an email, but no response. I got some issues of building cocos2d-x by mingw. Reply my email though your `sina mailbox`?

**Comment by akof1314** [comment]
@dumganhar Sorry so long to reply.

**Pull request merged**